src/clients/AcrossConfigStoreClient.ts

```diff
@@ -208,27 +222,53 @@ export class AcrossConfigStoreClient {
     return this.configStoreVersion >= version;
   }
 
-  async update(): Promise<void> {
+  async _update(): Promise<ConfigStoreUpdate> {
+    const latestBlockNumber = await this.configStore.provider.getBlockNumber();
     const searchConfig = {
       fromBlock: this.firstBlockToSearch,
-      toBlock: this.eventSearchConfig.toBlock || (await this.configStore.provider.getBlockNumber()),
+      toBlock: this.eventSearchConfig.toBlock || latestBlockNumber,
```

Comment:
Is there a reason for this modification?

Reply:
Recording the `latestBlockNumber` is a common pattern from the HubPoolClient and SpokePoolClient classes. I had been using it during test, but ultimately found that it wasn't required. I've backed it out from this PR now, since there's not strictly any need for it.

Reply:
Actually, it's needed for the Mocked variant, where the latest block number needs to be retained between updates. I've restored it.